### PR TITLE
Update data types and docs

### LINKED ISSUE #36: Clean up data types
Change data type from `String` to `Stdlib::Absolutepath`:

* `master`
* `map_dir`
* `mapfile`
* `mount`

Mark any Optional parameters as such with the Optional data type.

### PR DESCRIPTION
**Pull request by dhollinger**
closes #36 and #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,7 +101,6 @@ The direct and execute parameters allow for the creation of indirect mounts, see
 
 In hiera, there's a `autofs::mounts` class you can configure, for example:
 ```yaml
----
```
Comment (bastelfreak) [inline README.md]: Why did you remove this? I thought three dashes (not four) are totally valid in yaml?

**Comment by bastelfreak** [comment]
that reminds me that we somehow should automate the puppet-strings docs generation.

File: README.md (hunk 1 above)
Comment (dhollinger) [inline README.md]: My markdown preview tool was not formatting the HTML correctly when the dashes were there.

You're right, though, that is valid yaml

Comment (dhollinger) [inline README.md]: My markdown preview tool was not formatting the HTML correctly when the dashes were there.

You're right, though, that is valid yaml
